torch/_higher_order_ops/scan.py

```diff
@@ -110,9 +227,6 @@ def add(x: torch.Tensor, y: torch.Tensor):
     if not isinstance(reverse, bool):
         raise RuntimeError("Reverse must be a bool, but got " + str(type(reverse)))
 
-    # TODO: Support closures/nn_modules in order to be able represent RNNs with scan
```

Comment:
Don't need to happen in this PR: it seems to me that can we handle the dim kwarg in scan (i.e. the user-facing wrapper) by shift it backward to dim 0 so that we could remove all the dim arguments in scan_op. This would simplify everything related to dim. WDYT? Maybe add a TODO?

Reply:
I am not so sure, it may be that the `combine_fn` depends internally on the shape that the user provides, right?
One concrete example:
```
W = torch.randn(3, 5)
xs = torch.randn(5, 7, 3)
init = torch.randn(3, 3)
def f(x, y):
  return W @ x + y, W @ x + y
carry, out = scan(f, init, xs, 1)
```
In this case we cannot always move the dim to 0, as the `matmul` would break, right?

Reply:
like we could do scan(f, init, xs_shifted, 0), where xs_shifted.shape = (7, 5, 3)

Reply:
True, I'll add a TODO for that